chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -188,16 +188,17 @@
   action :create
 end
 
+neutron_server = search(:node, "run_list_map:neutron-server").first
 template "/etc/neutron/lbaas-connection.conf" do
   source "lbaas-connection.conf"
   mode "0640"
   owner "root"
   group "root"
   action :create
   variables(
-    lazy do
-      { sql_connection: node[:neutron][:db][:sql_connection] }
-    end
+    lazy {
+      { sql_connection: neutron_server[:neutron][:db][:sql_connection] }
+    }
   )
   only_if { roles.include? "neutron-network" }
```

Comment:
I think you need to make sure that `neutron_server` exists (`only_if { roles.include?("neutron-network") && !neutron_server.nil? }`